### PR TITLE
Upgrade with new msg based fees

### PR DESCRIPTION
**Pull request by channa-figure**
## Description

closes: #616 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: app/upgrades_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	app *App
+	ctx sdk.Context
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.app = Setup(false)
+	s.ctx = s.app.BaseApp.NewContext(false, tmproto.Header{})
+}
+
+func (s *IntegrationTestSuite) TestMarkerProposals() {
```
Comment (arnabmitra) [inline app/upgrades_test.go:28]: nice!

**Review by arnabmitra — APPROVED** [review]
lgtm

**Review by derekadams — APPROVED** [review]

**Review by arnabmitra — DISMISSED** [review]

File: app/upgrades.go [hunk 2]
```diff
@@ -46,12 +52,47 @@ var handlers = map[string]appUpgrade{
...
 			versionMap := app.UpgradeKeeper.GetModuleVersionMap(ctx)
 			ctx.Logger().Info("NOTICE: Starting migrations. This may take a significant amount of time to complete. Do not restart node.")
-			return app.mm.RunMigrations(ctx, app.configurator, versionMap)
+			resultVM, err := app.mm.RunMigrations(ctx, app.configurator, versionMap)
+			if err != nil {
+				return resultVM, err
+			}
+
+			if err := AddMsgBasedFees(app, ctx); err != nil {
+				return resultVM, err
```
Comment (iramiller) [inline app/upgrades.go:61]: this conditional and the return that follows are redundant ...

**Review by iramiller — APPROVED** [review]
overall looks good ... small nit on the error handling flow, could simplify.

**Review by arnabmitra — APPROVED** [review]

**Review by derekadams — APPROVED** [review]

**Review by iramiller — APPROVED** [review]
LGTM

**Pull request merged**